pennylane/transforms/split_non_commuting.py

```diff
@@ -28,6 +33,51 @@
 from pennylane.typing import PostprocessingFn, Result, ResultBatch, TensorLike
 
 
+@dataclass
+class SingleTermMP:
+    """A dataclass to represent a single-term observable in a measurement process."""
+
+    indices: list[int]
+    coeffs: list[complex]
+    group_idx: int
+    idx_in_group: int
+
+
+ShotDistributionFunction = Callable[[int, list[list[TensorLike]], int], Sequence[int]]
+
+
+def _uniform_deterministic_sampling(total_shots, coeffs_per_group, seed):
+    # pylint: disable=unused-argument
+    num_groups = len(coeffs_per_group)
+    shots = total_shots / num_groups
+    shots_per_group = np.array([shots] * num_groups)
+    return shots_per_group
+
+
+def _weighted_deterministic_sampling(total_shots, coeffs_per_group, seed):
+    # pylint: disable=unused-argument
+    sum_per_group = np.array([np.sum(np.abs(coeffs)) for coeffs in coeffs_per_group])
+    prob_shots = sum_per_group / np.sum(sum_per_group)
+    shots_per_group = total_shots * prob_shots
+    return shots_per_group
+
+
+def _weighted_random_sampling(total_shots, coeffs_per_group, seed):
+    np.random.seed(seed)
+    sum_per_group = np.array([np.sum(np.abs(coeffs)) for coeffs in coeffs_per_group])
+    prob_shots = sum_per_group / np.sum(sum_per_group)
+    distribution = multinomial(n=total_shots, p=prob_shots)
```

Comment:
```suggestion
    sum_per_group = np.array([np.sum(np.abs(coeffs)) for coeffs in coeffs_per_group])
    prob_shots = sum_per_group / np.sum(sum_per_group)
    distribution = multinomial(n=total_shots, p=prob_shots, seed=seed)
```
This keeps the randomness local so we don't interact with other random number generation in the system.

It also allows us to provide a random number generator to the seed to generate different results each call.